src/main/java/gov/nasa/pds/objectAccess/TableReader.java

```diff
@@ -311,6 +310,42 @@ public int getCurrentRow() {
 	public ByteWiseFileAccessor getAccessor() {
 	  return this.accessor;
 	}
+
+    private long countCarriageReturn(URL dataFile) throws Exception {
```

Comment:
@qchaupds just curious, when we do content validation, aren't we using this ByteBuffer then as well? In which case, doesn't this mean we are now going through and reading in the entire file multiple times? Could we combine this with content validation? It seems like if we have to open up the actual product to get some information (e.g. record size), then that could be considered in the realm of content validation.